eclipse-che/che-release
# chore: switch from configbump as a branch creation step to a release step; fix inconsistent ordering in make-release.sh; update readme for new phase order; simplify redundant text

### What does this PR do?

switch from configbump as a branch creation step to a release step; fix inconsistent ordering in make-release.sh
update readme for new phase order; simplify redundant text

Change-Id: I3c328d21aee05f49667aada35d4bc02f470e2992
Signed-off-by: Nick Boldt <nboldt@redhat.com>

### Screenshot/screencast of this PR
N/A

### What issues does this PR fix or reference?

Depends on https://github.com/che-incubator/configbump/pull/98

Fixes https://github.com/eclipse/che/issues/21988

### How to test this PR?

* Merge https://github.com/che-incubator/configbump/pull/98
* run the new action from https://github.com/che-incubator/configbump/actions
* see TODO in https://github.com/eclipse-che/che-release/pull/80 which we can then update and merge before 7.75.0 release day

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
